Domain/Services/Implementations/ReportService.cs

```diff
@@ -24,38 +25,63 @@ public ReportType GetByGuid<ReportType>(Guid guid)
         return report;
     }
 
-    public ReportType CreateReport<TargetType, ReportType>(Guid targetId, string title,
-        string details, ReportCategory category)
-        where TargetType : BaseEntity
-        where ReportType : Report
+    public Report GetByGuid(Guid guid)
+        => GetByGuid<Report>(guid);
+
+    public IEnumerable<Report> GetReports(ICollection<ReportType> types, bool open, bool closed)
     {
-        AllowOnlyLoggedIn();
+        AllowOnlyUser(ActingUser);
```

Comment:
This changes nothing. AllowOnlyLoggedIn enforces ActingUser not to be null. AllowOnlyUser checks whether ActingUser is equal to passed in User (here also ActingUser), so results are same.

Reply:
You are right, that doesn't make much sense to check if object is equal to itself.